### PR TITLE
Print map and generate map

### LINKED ISSUE #73: PrintMap()

### LINKED ISSUE #76: generateMap();

### PR DESCRIPTION
**Pull request by lukasz-kukulka**

### CODE REVIEW

**Review by BarTes8 — APPROVED** [review]

File: shm/src/Game.cpp [hunk 1]
```diff
@@ -10,23 +11,26 @@ Game::Game(size_t money, size_t game_days, size_t final_goal)
     , game_days_(game_days)
     , final_goal_(final_goal)
     , time_(std::make_unique<Time>())
+    , map_(std::make_unique<Map>())
 {
     current_day_ = time_->getElapsedTime();
     ship_ = std::make_unique<Ship>(1, 25, 100, nullptr);
     player_ = std::make_unique<Player>(std::move(ship_), 100, 10000);
+    mapGenerate();
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: No need to generate a map - it is already generated.

File: shm/inc/Game.hpp [hunk 2]
```diff
@@ -24,7 +30,10 @@ class Game {
     std::unique_ptr<Player> player_;
     std::unique_ptr<Time> time_;
     std::unique_ptr<Ship> ship_;
+    std::unique_ptr<Map> map_;
+    std::map<size_t, std::pair<size_t, size_t>> island;
```
Comment (mariuszlisowski) [inline shm/inc/Game.hpp]: no need for additional memory usage - container of islands already exists in Map class

File: shm/src/Game.cpp [hunk 3]
```diff
@@ -62,3 +66,21 @@ void Game::printMenu() {
...
+    size_t index = 1;
+    for (size_t i = 0; i <= COORDINATE_MAX; i++) {
+        for (size_t j = 0; j <= COORDINATE_MAX; j++) {
+            if (map_->getIsland(Island::Coordinates(i, j)) != nullptr) {
+                island.insert({index, std::make_pair(i, j)});
+                index++;
+            }
+        }
+    }
+}
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: redundant - generated map exists in Map class

File: shm/inc/Game.hpp [hunk 4]
```diff
@@ -16,6 +20,8 @@ class Game {
 private:
     void printWelcomeScreen();
     void printMenu();
+    void mapGenerate();
```
Comment (mariuszlisowski) [inline shm/inc/Game.hpp]: redundant

File: shm/src/Game.cpp [hunk 5]
```diff
@@ -62,3 +66,21 @@ void Game::printMenu() {
...
+
+void Game::printMap() {
+    for (const auto& [key, value] : island) {
```
Comment (mariuszlisowski) [inline shm/src/Game.cpp]: access map from Map class directly instead rewriting it to std::map

**Review by mariuszlisowski — CHANGES_REQUESTED** [review]

**Review by mariuszlisowski — COMMENTED** [review]
Fixes done in printMap_fix branch to be merged here and clean the comments later

**Review by mariuszlisowski — APPROVED** [review]

**Pull request merged**